src/engine/DnCManager.cpp

```diff
@@ -147,14 +150,15 @@ void DnCManager::solve( unsigned timeoutInSeconds )
 
     // Wait until either all subQueries are solved or a satisfying assignment is
     // found by some worker
-    while ( _numUnsolvedSubQueries.load() > 0 &&
-            !shouldQuitSolving.load() &&
-            !_timeoutReached )
+    while ( !shouldQuitSolving.load() && !_timeoutReached )
```

Comment:
This seems a little redundant. Why check _timeoutReached, if when it is reached we set shouldQuitSolving to true? Seems like it would suffice to check shouldQuitSolving